dev-tools/mage/integtest_kind.go

```diff
@@ -0,0 +1,143 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+// KindIntegrationTestStep setups a kind environment.
+type KindIntegrationTestStep struct{}
+
+// Name returns the kind name.
+func (m *KindIntegrationTestStep) Name() string {
+	return "kind"
+}
+
+// Use always returns false.
+//
+// This step should be defined in `StepRequirements` for the tester, for it
+// to be used. It cannot be autodiscovered for usage.
+func (m *KindIntegrationTestStep) Use(dir string) (bool, error) {
+	return false, nil
+}
+
+// Setup ensures that a kubernetes cluster is up and running.
+//
+// If `KUBECONFIG` is already deinfed in the env then it will do nothing.
+func (m *KindIntegrationTestStep) Setup(env map[string]string) error {
+	_, exists := env["KUBECONFIG"]
```

Comment:
It could be nice to add a README into the kubernetes module with brief instructions about the use of `KUBECONFIG` for local testing. Maybe with instructions for manually starting a local cluster with kind or minikube.

Reply:
Added a readme explaining how to run the integration tests for kubernetes.